src/playwright/testWithFixtures.ts

```diff
@@ -0,0 +1,57 @@
+// Disable linter rule that erroneously thinks Playwright's use function is a React hook:
+/* eslint-disable react-hooks/rules-of-hooks */
+import { test as base } from '@playwright/test';
+
+import { AccessibilityStatementCmsPage } from './pages/accessibilityStatementCms.page';
+import { CmsPage } from './pages/cms.page';
+import { FrontPage } from './pages/front.page';
+import { SearchPage } from './pages/search.page';
+
+type TestFixtures = {
+  // Base fixtures
+  frontPage: FrontPage;
+  cmsPage: CmsPage;
+  accessibilityStatementCmsPage: AccessibilityStatementCmsPage;
+  searchPage: SearchPage;
+  // Parametrized versions of base fixtures
+  defaultFrontPage: FrontPage;
+  fiFrontPage: FrontPage;
+  svFrontPage: FrontPage;
+  enFrontPage: FrontPage;
+};
+
+export const test = base.extend<TestFixtures>({
+  frontPage: async ({ page }, use) => {
+    await use(new FrontPage(page));
+  },
+
+  cmsPage: async ({ page }, use) => {
+    await use(new CmsPage(page));
+  },
+
+  accessibilityStatementCmsPage: async ({ page }, use) => {
+    await use(new AccessibilityStatementCmsPage(page));
+  },
+
+  searchPage: async ({ page }, use) => {
+    await use(new SearchPage(page));
+  },
+
+  defaultFrontPage: async ({ page }, use) => {
+    await use(await FrontPage.create(page, '/'));
+  },
+
+  fiFrontPage: async ({ page }, use) => {
+    await use(await FrontPage.create(page, '/fi'));
+  },
+
+  svFrontPage: async ({ page }, use) => {
+    await use(await FrontPage.create(page, '/sv'));
+  },
+
+  enFrontPage: async ({ page }, use) => {
+    await use(await FrontPage.create(page, '/en'));
+  },
```

Comment:
Instead of prefixing, I would append the locale to the end of the page name. It's easier to maintain and read when same page (in different language) is grouped with all it's options:

```suggestion
  frontPage: async ({ page }, use) => {
    await use(await FrontPage.create(page, '/'));
  },

  frontPageInFi: async ({ page }, use) => {
    await use(await FrontPage.create(page, '/fi'));
  },

  frontPageInSv: async ({ page }, use) => {
    await use(await FrontPage.create(page, '/sv'));
  },

  frontPageInEn: async ({ page }, use) => {
    await use(await FrontPage.create(page, '/en'));
  },
```

Reply:
Changed fixtures names in 0485f29485a5740ef82f878acf04a797b890d0f3